pkg/server/api/middleware/audit.go

```diff
@@ -0,0 +1,130 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/shirou/gopsutil/process"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/peertracker"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/pkg/server/api/audit"
+	"github.com/spiffe/spire/pkg/server/api/rpccontext"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func WithAuditLog(udsTrackerEnabled bool) Middleware {
+	return auditLogMiddleware{
+		udsTrackerEnabled: udsTrackerEnabled,
```

Comment:
The maintainers had a discussion on this today and decided that for now we'd like to see the auditing feature either be fully enabled or not enabled at all. Allowing this "half-enabled" auditing mode for the TCP endpoint only feels kind of counterintuitive because the auditing of certain requests becomes dependent on how the Server is configured.

Since audit logging is disabled by default, I think we should be ok from a compatibility perspective right now.

Reply:
I don't think this flag is needed anymore?

Reply:
it is for unit tests, to avoid reading tracker when it will not be possible/desirable on tests

Reply:
I see, that makes sense.